### ISSUE TITLE
Should be able to get a Compute Service without knowing its catalogName

### ISSUE THREAD

**Issue by fifieldt** [rackspace/php-opencloud#629]
Hi,

Currently, to get a compute service, you must run:

$service = $client->computeService('{catalogName}', '{region}', '{urlType}');

with catalogName (the name of the service as it appears in the service catalog) as a required option.

However:

* most users have no idea what the catalogName of their compute service is
* the catalogName is not displayed in the OpenStack dashboard
* the catalogName is not listed in the OpenStack RC file
* the catalogName is not listed when running 'keystone catalog' on the cli
* most OpenStack deployments have only one compute service per region

Currently, if catalogName is left blank, a string 'cloudServersOpenStack' is used, which does not work for most clouds.

Instead, I think the desired behaviour should be to check the catalog and if only one compute service is found, that should be the one returned.

**Comment by jamiehannaford** [comment]
@fifieldt I agree with you, but what happens in the rare situation where somebody has multiple compute entries?

**Comment by fifieldt** [comment]
Hi, unfortunately just changing to 'nova' doesn't help much. Though "nova" is used in the install guide, this is a freetext string you set when configuring the cloud - the last 3 clouds I looked at I had "Compute Service", "OpenStack Compute" and "nova".

Surely there's some logic where we make it an optional string, (or fix the logic for the null string case) and fail out with an error if there's multiple compute services in the region and the catalogName isn't specified?

This is kinda unusable at the moment :)

**Comment by jamiehannaford** [comment]
Yeah, so I think the desired behaviour would be:

* If the user doesn't supply a catalogName, go through the catalog and pick the only Nova service available. If multiple or none are found, raise an error. 
* If the user _does_ supply a catalogName, and no entry with that name exists, raise an error.

**Comment by fifieldt** [comment]
sounds perfect to me :)
